### DISCUSSION TITLE
Using a stackPanel to display a list of images that can be updated from the ModelView Dynamically

### DISCUSSION THREAD

**Discussion by Skyhikeeper** [AvaloniaUI/Avalonia#10045] [General]
I am now getting the hang of this but have encountered an issue, I want to display a list of images in a vertical view using a stackpanel, what is the best way to achieve this?

**Comment by lukaszossowski** [comment]
You are not very specific.

Create an ItemsControl and bind the collection of your images to it. You can customise the way ItemsControl displays its items via its ItemsPanel property. To have them automatically update the ViewModel should have the pictures in an ObservableCollection, which automatically calls property changed event when items are added or removed.
